content/08-cr-features.md

```diff
@@ -585,6 +585,54 @@ Whether a loop is unrolled or not depends on the loop unrolling cost, which is d
 
 
 
+<!--label:cr-tail-recursion-elimination-->
+### Tail Recursion Elimination (TRE)
+
+##### since Haxe 4.1.0
+
+Recursion is a natural way to solve some tasks like calculating fibonacci numbers or traversing trees. It's easy to write,
+read and understand. But it takes more computational resources at run time than solving the same tasks with loops instead of recursion.
+
+For example a loop-based function to find the root of a tree of nodes:
+```haxe
+  static function getRoot(node:Node):Node {
+    var root = node;
+    while(root.parent != null) {
+      root = root.parent;
+    }
+    return root;
+  }
+```
+Its recursive counterpart may look cleaner, but has worse performance because of additional function calls:
+```haxe
+  static function getRoot(node:Node):Node {
+    if(node.parent == null) {
+      return node;
+    }
+    return getRoot(node.parent);
+  }
+```
+
+Luckily Haxe compiler since version 4.1.0 can perform automatic recursion-to-loop transformation for functions which end with a recursive call.
+This optimisation is automatically enabled with the [static analyzer](/manual/cr-static-analyzer.html) (by adding `-D analyzer-optimize` to compilation arguments).
```

Comment:
As much as I like BrEng, I think we are consistent with American spelling in the manual. "optimization".